pkg/router/template/util/haproxy/map_entry.go

```diff
@@ -95,10 +95,19 @@ func generateSNIPassthroughMapEntry(cfg *BackendConfig) *HAProxyMapEntry {
 // generateCertConfigMapEntry generates a cert config map entry.
 func generateCertConfigMapEntry(cfg *BackendConfig) *HAProxyMapEntry {
 	if len(cfg.Host) > 0 && (cfg.Termination == routev1.TLSTerminationEdge || cfg.Termination == routev1.TLSTerminationReencrypt) && cfg.HasCertificate {
-		return &HAProxyMapEntry{
+		e := &HAProxyMapEntry{
 			Key:   fmt.Sprintf("%s.pem", cfg.Name),
 			Value: templateutil.GenCertificateHostName(cfg.Host, cfg.IsWildcard),
 		}
+		if cfg.HasCertificate {
+			switch cfg.Termination {
+			case routev1.TLSTerminationEdge:
+				fallthrough
+			case routev1.TLSTerminationReencrypt:
```

Comment:
Better to use a single case:
```go
			case routev1.TLSTerminationEdge, routev1.TLSTerminationReencrypt:
				e.SSLBindConfig = "[alpn h2,http/1.1]"
```

Reply:
Done in https://github.com/openshift/router/pull/125/commits/e604cfad701a5e481d34379cfb62647f0f09558f